### DISCUSSION TITLE
Errors installing on Mac M2 and Ubuntu 24

### DISCUSSION THREAD

**Discussion by gabruni** [jive-vlbi/sched#29] [Installation]
Hi,

I get the following error when trying to install pySCHED on my macbook air M2 or in my Ubuntu 24 machine, thourgh the conda environment command (yaml file):

```
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [152 lines of output]
      /private/var/folders/sr/wnn9tf_50n3gs3l0sfk7tjxr0000gn/T/pip-install-hcxfljrh/pythonsched_eb61a51d71df4f35b9671b6b52ff413c/setup.py:5: DeprecationWarning:
      
        `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
        of the deprecation of `distutils` itself. It will be removed for
        Python >= 3.12. For older Python versions it will remain present.
        It is recommended to use `setuptools < 60.0` for those Python versions.
        For more details, see:
          https://numpy.org/devdocs/reference/distutils_status_migration.html
   

```

I get the same error if I try with the pip3 command on my mac, while if I try the pip3 command on Ubuntu I get yet another one:

```
Building wheels for collected packages: pythonSCHED
  Building wheel for pythonSCHED (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [364 lines of output]
```

Any help? Thanks
Gabriele

**Comment by bobeldering** [comment]
Hi Gabriele,

I have updated the .yaml file (available [here](https://raw.githubusercontent.com/jive-vlbi/sched/refs/heads/python/pySCHED.yaml)). This new version restricts the version of python (<3.12) and setuptools (<60), as that introduced deprecation to the distutils. We will look into fixing this properly, but I hope this at least allows you to install it right now through conda.

Cheers,
Bob

**Comment by semonyy** [comment]
I also get the same problem in Macbook and conda, it don't work it with new Yaml file.

**Comment by semonyy** [comment]
simon@simondeMacBook-Pro-2 Downloads % conda env create -f pySCHED.yaml

Channels:
 - conda-forge
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: done
Installing pip dependencies: | Ran pip subprocess with arguments:
['/Users/simon/miniforge3/envs/pySCHED/bin/python', '-m', 'pip', 'install', '-U', '-r', '/Users/simon/Downloads/condaenv.kktdxjhh.requirements.txt', '--exists-action=b']
Pip subprocess output:
Collecting pythonSCHED (from -r /Users/simon/Downloads/condaenv.kktdxjhh.requirements.txt (line 1))
  Using cached pythonSCHED-1.28.3.tar.gz (7.0 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'

Pip subprocess error:
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/Users/simon/miniforge3/envs/pySCHED/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
          main()
        File "/Users/simon/miniforge3/envs/pySCHED/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 373, in main
          json_out["return_val"] = hook(**hook_input["kwargs"])
        File "/Users/simon/miniforge3/envs/pySCHED/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 143, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/mf/dkk_f4v96v189d___s2p10_80000gn/T/pip-build-env-y5y6m61k/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 331, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
        File "/private/var/folders/mf/dkk_f4v96v189d___s2p10_80000gn/T/pip-build-env-y5y6m61k/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 301, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/mf/dkk_f4v96v189d___s2p10_80000gn/T/pip-build-env-y5y6m61k/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 512, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/private/var/folders/mf/dkk_f4v96v189d___s2p10_80000gn/T/pip-build-env-y5y6m61k/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 317, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
      ModuleNotFoundError: No module named 'numpy'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed to build 'pythonSCHED' when getting requirements to build wheel

failed

CondaEnvException: Pip failed

**Reply by bobeldering** [reply]
I currently don't have a Mac available for testing, but just looking at your log output:

`ModuleNotFoundError: No module named 'numpy'
`
This line suggests that numpy was not installed properly as part of the conda environment.

To check this, you could remove the pySCHED installation from the .yaml file (edited file attached: 
[pySCHED-test.yaml](https://github.com/user-attachments/files/24099116/pySCHED-test.yaml)), activate the environment and then check the version and location of numpy. Something like this:

```
$ conda env create -f pySCHED-test.yaml
$ conda activate pySCHED-test
(pySCHED-test) $ python                                                                                                                                  
Python 3.11.14 | packaged by conda-forge | (main, Oct 22 2025, 22:46:25) [GCC 14.3.0] on linux                                                                                  
Type "help", "copyright", "credits" or "license" for more information.                                                                                                          
>>> import numpy
>>> numpy.__version__, numpy.__file__
('1.23.5', '/home/bobeldering/miniforge3/envs/pySCHED-test/lib/python3.11/site-packages/numpy/__init__.py')                                                                     
>>> 
```

If this shows numpy version 1.23.x, you could try to install pySCHED in the pySCHED-test environment manually with extra debug to see what happens:
`(pySCHED-test) $ pip install -vvv pythonSCHED |& tee pySCHED-install.log`

If this still fails, the content of the pySCHED-install.log file might be helpful to post.

**Comment by bobeldering** [comment]
I turned out pip version >= 25.3 exposes this problem in the pySCHED build system. See #32 , the quick fix suggest by @ptiede has been merged into the [pySCHED.yaml](https://github.com/jive-vlbi/sched/raw/refs/heads/python/pySCHED.yaml) file.
